File: app/src/main/java/com/przemolab/oknotifier/modules/NotifierRepositoryModule.java
package com.przemolab.oknotifier.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class NotifierRepositoryModule {

    private Context context;

    public NotifierRepositoryModule(Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    public NotifierRepository provideNotifierRepository() {
        return new NotifierRepository(context);
    }
}

File: app/src/main/java/com/przemolab/oknotifier/modules/NotifierRepositoryModule.kt
package com.przemolab.oknotifier.modules

import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
open class NotifierRepositoryModule(private val context: Context) {

    @Provides
    @Singleton
    open fun provideNotifierRepository(): NotifierRepository {
        return NotifierRepository(context)
    }
}
